library-aux/src/scala/AnyKind.scala

```diff
@@ -0,0 +1,7 @@
+package scala
+
+/** The super-type of all types.
+ *
+ *   See [[https://docs.scala-lang.org/scala3/reference/other-new-features/kind-polymorphism.html]].
+ */
+final abstract class AnyKind
```

Comment:
It makes no sense for this type to be `final`. It definitely has subclasses, including `AnyRef`.
```suggestion
abstract class AnyKind
```

Reply:
I based it on `Definitions.scala`:
https://github.com/scala/scala3/blob/75a15c2131d67bb025701e1193573dc5d3a9a3ab/compiler/src/dotty/tools/dotc/core/Definitions.scala#L450-L451

When trying to extend `AnyKind`, the following errors are reported:
```
$ scala-cli repl            
Welcome to Scala 3.4.2 (11, Java OpenJDK 64-Bit Server VM).
Type in expressions for evaluation. Or try :help.
                                                                                                                                                                                                                                                   
scala> class Test extends AnyKind
-- [E093] Syntax Error: --------------------------------------------------------
1 |class Test extends AnyKind
  |      ^
  |      class Test cannot extend final class AnyKind
  |
  | longer explanation available when compiling with `-explain`
-- Error: ----------------------------------------------------------------------
1 |class Test extends AnyKind
  |                   ^^^^^^^
  |                   AnyKind does not have a constructor
2 errors found
```

Reply:
Definitions is what works for the compiler with the minimum amount of work. From a spec point of view, `AnyKind` is not even a class (nor a trait). It's a completely unique thing. The error message that you mention is definitely not a good message. We shouldn't tailor the doc to that poor message. We should tailor it to the real spec as much as possible instead. 

(Arguably it should even be an abstract type as well, but since `Nothing` is presented as a class, it makes "less not" sense to also use a class here.)

Reply:
The docs even say "It is declared abstract and final"